### PR TITLE
add libchannelserver

### PR DESCRIPTION
**Pull request by bxq2011hust**

### CODE REVIEW

File: libchannelserver/ChannelException.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+    This file is part of cpp-ethereum.
```
Comment (wheatli) [inline libchannelserver/ChannelException.h]: Plz make the header COPYRIGHT as same as other modules~~

File: libchannelserver/ChannelException.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file: ChannelException.h
+ * @author: fisco-dev
```
Comment (wheatli) [inline libchannelserver/ChannelException.h]: use your own name, and change date

File: libchannelserver/ChannelException.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    ChannelException(){};
+    ChannelException(int errorCode, const std::string& msg) : _errorCode(errorCode), _msg(msg){};
+
+    virtual int errorCode() { return _errorCode; };
+    virtual const char* what() const _GLIBCXX_USE_NOEXCEPT override { return _msg.c_str(); };
+
+    bool operator!() const { return _errorCode == 0; }
+
+private:
+    int _errorCode = 0;
```
Comment (wheatli) [inline libchannelserver/ChannelException.h]: change the code style to suitable for new standard

File: libchannelserver/ChannelServer.cpp [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+            boost::bind(&ChannelServer::onAccept, shared_from_this(),
+                boost::asio::placeholders::error, session));
+    }
+    catch (std::exception& e)
+    {
+        LOG(ERROR) << "ERROR:" << e.what();
+    }
+}
+
+#if 0
```
Comment (wheatli) [inline libchannelserver/ChannelServer.cpp]: delete useless codes, make sure all the codes protected by "#if 0" are droped

File: libchannelserver/ChannelServer.cpp [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+    }
+
+    try
+    {
+        LOG(DEBUG) << "Close ioService";
+        _ioService->stop();
+    }
+    catch (std::exception& e)
+    {
+        LOG(ERROR) << "ERROR:" << e.what();
```
Comment (wheatli) [inline libchannelserver/ChannelServer.cpp]: Would you like to define a macro like CHANNEL_LOG, and tag some special flag for channel server's log output

File: libchannelserver/ChannelRPCServer.cpp [hunk 6]
```diff
@@ -0,0 +1,1055 @@
...
+        LOG(ERROR) << "decode error:" << result << " package size:" << message->size();
+        return;
+    }
+
+    LOG(DEBUG) << "receive node mssage length:" << message->size() << " type:" << msg->type()
+               << " seq:" << msg->seq();
+
+    switch (msg->type())
+    {
+#if 0
```
Comment (wheatli) [inline libchannelserver/ChannelRPCServer.cpp]: delete all theses useless codes, make sure there is no chinese code or comment

File: libchannelserver/ChannelException.h (hunk 1 above)
Comment (bxq2011hust) [reply libchannelserver/ChannelException.h]: Will fix it later

File: libchannelserver/ChannelException.h (hunk 2 above)
Comment (bxq2011hust) [reply libchannelserver/ChannelException.h]: modified

File: libchannelserver/ChannelRPCServer.cpp (hunk 6 above)
Comment (bxq2011hust) [reply libchannelserver/ChannelRPCServer.cpp]: Ok

File: libchannelserver/ChannelServer.cpp (hunk 5 above)
Comment (bxq2011hust) [reply libchannelserver/ChannelServer.cpp]: Not for now. If channelServer will be deprecated, I think we should not spend much time on it?

File: libchannelserver/ChannelServer.cpp (hunk 4 above)
Comment (bxq2011hust) [reply libchannelserver/ChannelServer.cpp]: deleted

**Pull request merged**